New cleanup plugin that deletes offset files after tests (closes #669)

mnmelo opened the pull request.

kain88-de commented:
When is this plugin run? Does nose pick it up magically?

mnmelo commented:
Sorry, I realize some documentation would have helped!

As it is now you must pass the `--with-mda_cleanup` flag to `mda_nosetests`.
I realize I also didn't add the flag to the travis builds.

I don't have any strong feelings for it being on or off by default, or whether to run it in travis (where it is unneeded since state is not kept between builds). What say you?

kain88-de commented:
I think @jbarnoud wants it on by default.

jbarnoud commented:
I do not really mind if it is activated by default or not. On one hand, having it by default will avoid running into problems when the offset format changes, on the other hand, we may not notice these problems.

I know I will activate it in my own testing script at least.

kain88-de commented on testsuite/MDAnalysisTests/plugins/cleanup.py:
```diff
@@ -0,0 +1,52 @@
+        from .. import __name__ as mdatestsname
+        dirname = resource_filename(mdatestsname, 'data')
+        #print("Cleanup: looking for offsets to clean in {}".format(dirname))
+        for root, dirs, fnames in walk(dirname):
```
does this walk down directories?

jbarnoud replied:
Yes it does.

mnmelo commented:
Ok, I made it on by default and cleaned up other minor aspects.
It'll report what it's doing if nose's `verbosity` is greater than 2 (`-v` sets `verbosity=2`, so you'll need to specifically ask for `--verbosity=3` if you want to see anything.)

I also cleaned up minor things in the `error_capture` plugin. It might be a good idea to create our own subclass of Plugin that leaves a plugin on by default (right now `error_capture` and `cleanup` have some code duplication in this respect.)

And yes, it traverses subdirs. Here's the output on a testrun with `--verbosity=3`:
```
----------------------------------------------------------------------
Ran 3326 tests in 55.887s

OK (KNOWNFAIL=3, SKIP=5)
Cleanup: looking for offsets to clean in /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.gram_A_identical_frames.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.xtc_offsets.npz
```

kain88-de commented:
Can you change the verbosity to be enabled at level 1. Ideally level 0. We are doing something non-standard here and I think we should inform devs who use this later early no to avoid surprises for them.

mnmelo commented:
I can do it, but are you sure you want this to show up even when we're at low verbosity? I think it kills the simplicity of it and of having the summary as last output line. But I don't really care about it.
(I can also try making this show up before the summary).

This is how the non-verbose default would look like:
```
....S...................S....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K...............................K...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S..SS....................................................................................................................................................................................................................................................K...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 3326 tests in 56.123s

OK (KNOWNFAIL=3, SKIP=5)
Cleanup: looking for offsets to clean in /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.gram_A_identical_frames.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.xtc_offsets.npz
```

kain88-de commented:
I would activate it with simple verbosity.  That should be enough

mnmelo commented:
Ok, that sounds like a good compromise (and will make it show up in travis). Still, I'll try to move output before the summary line.

mnmelo commented:
ping @kain88-de Is this ok for merging? It's blocking the 0.14 release...

mnmelo commented:
Hmm, I just noticed that travis' verbosity changed from what I expected. I'm quickly updating the verbosity limits.

kain88-de commented:
Looks  good. Thanks for fixing it.
